src/rum/instrumentation/DdRumResourceTracking.tsx

```diff
@@ -5,14 +5,66 @@
  */
 
 import { DdRum } from '../../foundation'
-import type { DdRumXhr } from './DdRumXhr'
+import type { DdRumXhr, DdRumXhrContext } from './DdRumXhr'
 import { generateTraceId } from './TraceIdentifier';
 
 export const TRACE_ID_HEADER_KEY = "x-datadog-trace-id"
 export const PARENT_ID_HEADER_KEY = "x-datadog-parent-id"
 export const ORIGIN_HEADER_KEY = "x-datadog-origin"
 export const ORIGIN_RUM = "rum"
 
+const MISSING_TIME = -1
+
+type Duration = number & { d: 'Duration in ms' }
+type ServerDuration = number & { s: 'Duration in ns' }
+
+interface Timing {
+  startTime: ServerDuration
+  duration: ServerDuration
+}
+
+interface ResourceTimings {
+  // unlike in Performance API it is not the time until request
+  // starts (requestStart, before it can be connect, SSL, DNS),
+  // but the time until the response is first seen
+  firstByte: Timing,
+  download: Timing,
+}
+
+function createTimings(requestContext: DdRumXhrContext, responseEndTime: number): ResourceTimings | null {
+  const {
+    startTime,
+    loadStartTime
+  } = requestContext
+
+  if (startTime === MISSING_TIME || loadStartTime === MISSING_TIME) {
+    return null
+  }
+
+  const firstByte = formatTiming(startTime, startTime, loadStartTime)
+  const download = formatTiming(startTime, loadStartTime, responseEndTime)
+
+  return {
+    firstByte,
+    download
+  }
+}
+
+function formatTiming(origin: number, start: number, end: number): Timing {
+  return {
+    duration: toServerDuration((end - start) as Duration),
+    startTime: toServerDuration((start - origin) as Duration),
+  }
+}
+
+function toServerDuration(duration: Duration) {
+  return round(duration * 1e6, 0) as ServerDuration
+}
```

Comment:
Why not using `Math.round` function ?

Reply:
don't have answer here, this is what is used in `browser-sdk`